src/app/services/versions/versions.service.ts

```diff
@@ -0,0 +1,321 @@
+  }
+
+  get allVersionsOfCurrentIntegration$() {
+    return this.allVersionsOfCurrentIntegration$$.asObservable().pipe(filter((data) => data !== null));
+  }
+
+  get integrationId$() {
+    return this.integrationId$$.asObservable().pipe(filter((data) => data !== null));
+  }
+
+  constructor(
+    private integrationsClientService: IntegrationsClientService,
+    private categoryItemsClientService: CategoryItemsClientService,
+    private versionsClientService: VersionsClientService,
+    private categoryItemsService: CategoryItemsService,
+    private versionsComponentsCommunicationService: VersionsComponentsCommunicationService,
+    private notificationService: NotificationService,
+    private dialog: MatDialog
+  ) {
+    this.setupAddCategoryItemStream();
+    this.setupRemoveCategoryItemStream();
+  }
+
+  saveIntegrationId(id: number): void {
+    if (id) {
+      this.integrationId$$.next(id);
+    } else {
+      throw Error('Wrong integrationd ID!');
+    }
+  }
+
+  emitPrimaryIntegrationVersion() {
+    return this.getAndAssemblePrimaryIntegrationVersionWithCategoryItems().pipe(
+      tap((extendedIntegration) => this.integrationPrimaryVersion$$.next(extendedIntegration))
+    );
+  }
+
+  getAndAssemblePrimaryIntegrationVersionWithCategoryItems(): Observable<ExtendedIntegration> {
+    return this.integrationId$.pipe(
+      mergeMap((id) => this.integrationsClientService.getFullIntegrationById(id!)),
+      // it is mandatory, because BACKEND doesn't do this job
+      map((integration) => ({
+        ...integration!,
+        appliedVersion: integration.versions?.find((version) => !!version.isFinal),
+      })),
+      mergeMap((integration) =>
+        this.getCategoryItemsOfVersion(integration.appliedVersion?.id!).pipe(
+          map((categoryItems) => ({
+            ...createIntegrationWithCategoryItemsAndCategoryCost(integration, categoryItems),
+          }))
+        )
+      )
+    );
+  }
+
+  setSecondaryVersion(version: Version) {
+    if (version) {
+      this.secondaryVersion$$.next(version);
+    } else {
+      throw Error('Invalid Version ID!');
+    }
+  }
+
+  setSecondaryVersionStartingValue() {
+    const firstOfAllVersions = this.allVersionsOfCurrentIntegration$$.value?.[0];
+
+    if (firstOfAllVersions) {
+      this.secondaryVersion$$.next(firstOfAllVersions);
+    }
+  }
+
+  emitSecondaryIntegrationVersion() {
+    return this.assembleSecondaryIntegrationVersionWithCategoryItems().pipe(
+      tap((secondaryIntegrationVersion) => this.integrationSecondaryVersion$$.next(secondaryIntegrationVersion))
+    );
+  }
+
+  assembleSecondaryIntegrationVersionWithCategoryItems(): Observable<ExtendedIntegration> {
+    let secondaryVersion: Version;
+
+    return this.secondaryVersion$.pipe(
+      tap((version) => (secondaryVersion = version!)),
+      withLatestFrom(this.integrationPrimaryVersion$$),
+      map(([_, integration]) => ({ ...integration!, appliedVersion: secondaryVersion })),
```

Comment:
instead tap, withLatesFrom use combineLatest([this.secondaryVersion$, this.integrationPrimaryVersion$$])